docs/storage_snippets.py

```diff
@@ -0,0 +1,241 @@
+
+"""Testable usage examples for Google Cloud Storage API wrapper
+
+Each example function takes a ``client`` argument (which must be an instance
+of :class:`google.cloud.storage.client.Client`) and uses it to perform a task
+with the API.
+
+To facilitate running the examples as system tests, each example is also passed
+a ``to_delete`` list;  the function adds to the list any objects created which
+need to be deleted during teardown.
+"""
+
+import time
+
+from google.cloud import storage
+
+
+def snippet(func):
+    """Mark ``func`` as a snippet example function."""
+    func._snippet = True
+    return func
+
+
+def _millis():
+    return time.time() * 1000
+
+
+@snippet
+def storage_get_started(client, to_delete):
+    # [START storage_get_started]
+    from google.cloud import storage
+    client = storage.Client()
+    bucket = client.get_bucket('bucket-id-here')
+    # Then do other things...
+    blob = bucket.get_blob('/remote/path/to/file.txt')
+    print(blob.download_as_string())
+    blob.upload_from_string('New contents!')
+    blob2 = bucket.blob('/remote/path/storage.txt')
+    blob2.upload_from_filename(filename='/local/path.txt')
+    # [END storage_get_started]
+
+
+@snippet
+def client_bucket_acl(client, to_delete):
+    bucket_name = 'system-test-bucket'
+    bucket = client.bucket(bucket_name)
+    bucket.create()
+
+    # [START client_bucket_acl]
+    from google.cloud import storage
+    client = storage.Client()
+    bucket = client.get_bucket(bucket_name)
+    acl = bucket.acl
+    # [END client_bucket_acl]
+    to_delete.append(bucket)
+
+    # [START acl_user_settings]
+    acl.user('me@example.org').grant_read()
+    acl.all_authenticated().grant_write()
+    # [END acl_user_settings]
+
+    # [START acl_save]
+    acl.save()
+    # [END acl_save]
+
+    # [START acl_revoke_write]
+    acl.all().grant_read().revoke_write()
+    # [END acl_revoke_write]
+
+    # [START acl_save_bucket]
+    bucket.acl.save(acl=acl)
+    # [END acl_save_bucket]
+
+    # [START acl_print]
+    print(list(acl))
+    # [{'role': 'OWNER', 'entity': 'allUsers'}, ...]
+    # [END acl_print]
+
+
+@snippet
+def download_to_file(client, to_delete):
+    # [START download_to_file]
+    from google.cloud import storage
+    from google.cloud.storage import Blob
+
+    client = storage.Client(project='my-project')
+    bucket = client.get_bucket('my-bucket')
+    encryption_key = 'aa426195405adee2c8081bb9e7e74b19'
+    blob = Blob('secure-data', bucket, encryption_key=encryption_key)
+    with open('/tmp/my-secure-file', 'wb') as file_obj:
+        blob.download_to_file(file_obj)
+    # [END download_to_file]
+
+
+@snippet
+def upload_from_file(client, to_delete):
+    # [START upload_from_file]
+    from google.cloud import storage
+    from google.cloud.storage import Blob
+
+    client = storage.Client(project='my-project')
+    bucket = client.get_bucket('my-bucket')
+    encryption_key = 'aa426195405adee2c8081bb9e7e74b19'
```

Comment:
No they are all random, I just thought it'd be "better" to have two distinct random keys.